<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Biodiversity data</title>

<script src="site_libs/header-attrs-2.26/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/clipboard-1.7.1/clipboard.min.js"></script>
<link href="site_libs/primer-tooltips-1.4.0/build.css" rel="stylesheet" />
<link href="site_libs/klippy-0.0.0.9500/css/klippy.min.css" rel="stylesheet" />
<script src="site_libs/klippy-0.0.0.9500/js/klippy.min.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>





<style type="text/css">
/* for pandoc --citeproc since 2.11 */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Ecosystem dynamics and biodiversity</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Pracs: Spatial data
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="0_setup.html">0. Getting started</a>
    </li>
    <li>
      <a href="1_SpatialData.html">1. Spatial data in R</a>
    </li>
    <li>
      <a href="2_BiodivData.html">2. Biodiversity data in R</a>
    </li>
    <li>
      <a href="3_EnvData.html">3. Environmental data in R</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Pracs: Species distribution models
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="4_SDM_intro.html">4. SDMs: simple model fitting</a>
    </li>
    <li>
      <a href="5_SDM_eval.html">5. SDMs: assessment and prediction</a>
    </li>
    <li>
      <a href="6_SDM_algorithms.html">6. SDMs: algorithms and ensembles</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Pracs: population modelling
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="7_RS_intro.html">7. Getting started with RangeShiftR</a>
    </li>
    <li>
      <a href="8_RS_lynx.html">8. RangeShiftR: Eurasian lynx reintroduction</a>
    </li>
    <li>
      <a href="9_RS_grouse.html">9. RangeShiftR: Black grouse range dynamics</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Pracs: ecosystem modelling
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="10_Mad_intro.html">10. Running MadingleyR</a>
    </li>
    <li>
      <a href="11_Mad_carnivores.html">11. MadingleyR: Role of large carnivores</a>
    </li>
    <li>
      <a href="12_Mad_landuse.html">12. MadingleyR: land use change effects</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://www.uni-potsdam.de/en/ibb-macroecology/index">
    <span class="fa fa-home"></span>
     
  </a>
</li>
<li>
  <a href="https://twitter.com/ZurellLab">
    <span class="fa fa-twitter"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Biodiversity data</h1>

</div>


<script>
  addClassKlippyTo("pre.r, pre.markdown");
  addKlippy('right', 'top', 'auto', '1', 'Copy code', 'Copied!');
</script>
<hr />
<div class="alert alert-info">
<p><strong>RStudio project</strong></p>
<p>Open the RStudio project that we created in the first session. I
recommend to use this RStudio project for the entire course and within
the RStudio project create separate R scripts for each session.</p>
<ul>
<li>Create a new empty R script by going to the tab “File”, select “New
File” and then “R script”</li>
<li>In the new R script, type
<code># Session 2: Biodiversity data</code> and save the file in your
folder “scripts” within your project folder, e.g. as
“2_BiodivData.R”</li>
</ul>
</div>
<p>We are living in an age of big data where biodiversity data are
becoming increasingly available in digital format and at a global scale
<span class="citation">(Wüest et al. 2020)</span>. Many different types
of biodiversity data exist relating to the distribution of species
(e.g. from standardised monitoring schemes, citizen science platforms,
or expert knowledge), to species conservation status, species traits,
etc. Each of these data types comes with own challenges. Here, we simply
want to learn how we can obtain and process typical types of species
data. Specifically, we will work with range maps of terrestrial animals
and plants from the IUCN (Internation Union for the Conservation of
Nature, iucnredlist.org/) and the red list data from IUCN.</p>
<div id="range-maps" class="section level1" number="1">
<h1><span class="header-section-number">1</span> Range maps</h1>
<p>We rarely have detailed biodiversity data available over large
geographic extents. At broad (continental to global) extents,
expert-drawn range maps (also called extent-of-occurrence maps) are
often the primary data source on species distributions.</p>
<div id="iucn-range-maps" class="section level2" number="1.1">
<h2><span class="header-section-number">1.1</span> IUCN range maps</h2>
<p>The IUCN (the International Union for the Conservation of Nature)
provides expert range maps for a large number of species including
mammals, birds (through <a href="http://www.birdlife.org/">BirdLife
International</a>), amphibians, reptiles, and freshwater fishes. There
are also some range maps on plants and marine species, but these are
very limited taxonomically. Have a look for which taxa range maps are
available: <a
href="https://www.iucnredlist.org/resources/spatial-data-download"
class="uri">https://www.iucnredlist.org/resources/spatial-data-download</a>.
You can download them for free, but you should provide some information
on your work to obtain the data.</p>
<p>Most of the IUCN data are provided in the form of shapefiles. In
practical 2, we have already loaded the range map of the <a
href="data/IUCN_Sorex_alpinus.zip">Alpine shrew</a> and learned how to
use the package <code>terra</code> for reading in the shapefiles. You
may remember that the shapefile is recognized as <code>SpatVector</code>
object.</p>
<pre class="r"><code>library(terra)

# Load the shapefile
(shrew &lt;- terra::vect(&#39;data/IUCN_Sorex_alpinus.shp&#39;))</code></pre>
<pre><code>##  class       : SpatVector 
##  geometry    : polygons 
##  dimensions  : 1, 27  (geometries, attributes)
##  extent      : 5.733728, 26.67935, 42.20601, 51.89984  (xmin, xmax, ymin, ymax)
##  source      : IUCN_Sorex_alpinus.shp
##  coord. ref. : lon/lat WGS 84 (EPSG:4326) 
##  names       : id_no      binomial presence origin seasonal compiler yrcompiled
##  type        : &lt;chr&gt;         &lt;chr&gt;    &lt;int&gt;  &lt;int&gt;    &lt;int&gt;    &lt;chr&gt;      &lt;int&gt;
##  values      : 29660 Sorex alpinus        1      1        1     IUCN       2008
##         citation source dist_comm (and 17 more)
##            &lt;chr&gt;  &lt;chr&gt;     &lt;chr&gt;              
##  IUCN (Internat~     NA        NA</code></pre>
<pre class="r"><code># Plot the Central Europe
library(maps)
map(&#39;world&#39;,xlim=c(5,30), ylim=c(40,55))

# Overlay the range of the Alpine Shrew
plot(shrew, col=&#39;red&#39;, add=T)</code></pre>
<p><img src="2_BiodivData_files/figure-html/unnamed-chunk-1-1.png" width="672" /></p>
<p>Unfortunately, there is no API for the IUCN range maps. So, you need
to register with IUCN and then you can download range maps for different
taxonomic groups. The range maps for birds come as separate shape files
per species, while the range maps of mammals are provided as single
shapefile that contains all species. Course participants can download
the mammal shapefile from the moodle course - of course, the license
agreements by the IUCN apply!</p>
<pre class="r"><code># Read shapefile for all mammals using the raster package:
mammals &lt;- terra::vect(&#39;data/MAMMTERR.shp&#39;)</code></pre>
<p>The shapefile contains the range polygons for all described mammal
species. The attribute table contains information on species’ PRESENCE,
ORIGIN, and SEASONALity. Please have a look at the metadata to
understand the different values these attributes can be coded as.</p>
<pre class="r"><code>mammals</code></pre>
<pre><code>##  class       : SpatVector 
##  geometry    : polygons 
##  dimensions  : 43444, 14  (geometries, attributes)
##  extent      : -180.0001, 180.0025, -56.66595, 90  (xmin, xmax, ymin, ymax)
##  source      : MAMMTERR.shp
##  coord. ref. : lon/lat WGS 84 (EPSG:4326) 
##  names       :        BINOMIAL PRESENCE ORIGIN SEASONAL COMPILER  YEAR
##  type        :           &lt;chr&gt;    &lt;int&gt;  &lt;int&gt;    &lt;int&gt;    &lt;chr&gt; &lt;int&gt;
##  values      : Amblysomus cor~        1      1        1     IUCN  2008
##                Amblysomus hot~        1      1        1     IUCN  2008
##                Amblysomus mar~        1      1        1     IUCN  2008
##         CITATION SOURCE DIST_COMM ISLAND (and 4 more)
##            &lt;chr&gt;  &lt;chr&gt;     &lt;chr&gt;  &lt;chr&gt;             
##  IUCN (Internat~     NA        NA     NA             
##  IUCN (Internat~     NA        NA     NA             
##  IUCN (Internat~     NA        NA     NA</code></pre>
<pre class="r"><code># Inspect attribute table
terra::head(mammals)</code></pre>
<pre><code>##                     BINOMIAL PRESENCE ORIGIN SEASONAL COMPILER YEAR
## 1         Amblysomus corriae        1      1        1     IUCN 2008
## 2     Amblysomus hottentotus        1      1        1     IUCN 2008
## 3         Amblysomus marleyi        1      1        1     IUCN 2008
## 4        Amblysomus robustus        1      1        1     IUCN 2008
## 5 Amblysomus septentrionalis        1      1        1     IUCN 2008
## 6 Amblysomus septentrionalis        1      1        1     IUCN 2008
##                                                CITATION SOURCE DIST_COMM ISLAND
## 1 IUCN (International Union for Conservation of Nature)   &lt;NA&gt;      &lt;NA&gt;   &lt;NA&gt;
## 2 IUCN (International Union for Conservation of Nature)   &lt;NA&gt;      &lt;NA&gt;   &lt;NA&gt;
## 3 IUCN (International Union for Conservation of Nature)   &lt;NA&gt;      &lt;NA&gt;   &lt;NA&gt;
## 4 IUCN (International Union for Conservation of Nature)   &lt;NA&gt;      &lt;NA&gt;   &lt;NA&gt;
## 5 IUCN (International Union for Conservation of Nature)   &lt;NA&gt;      &lt;NA&gt;   &lt;NA&gt;
## 6 IUCN (International Union for Conservation of Nature)   &lt;NA&gt;      &lt;NA&gt;   &lt;NA&gt;
##   SUBSPECIES SUBPOP  SHAPE_area SHAPE_len
## 1       &lt;NA&gt;   &lt;NA&gt;  4.93503476 15.575123
## 2       &lt;NA&gt;   &lt;NA&gt; 19.94471634 28.392110
## 3       &lt;NA&gt;   &lt;NA&gt;  0.12779360  1.902574
## 4       &lt;NA&gt;   &lt;NA&gt;  0.09936845  1.655898
## 5       &lt;NA&gt;   &lt;NA&gt;  0.25763915  2.078154
## 6       &lt;NA&gt;   &lt;NA&gt;  2.70333866  7.080348</code></pre>
<p>We can search for specific species or species groups in the attribute
table in different ways:</p>
<pre class="r"><code># Range map for the species &#39;Lynx lynx&#39;
terra::subset(mammals, mammals$BINOMIAL == &quot;Lynx lynx&quot;)

# Show all entries for the species with the word &#39;Lynx&#39; in their name
grep(&#39;Lynx&#39;,mammals$BINOMIAL, value=T)

# Range map for all species with the word &#39;Lynx&#39; in their name
mammals[grep(&#39;Lynx&#39;, mammals$BINOMIAL),]</code></pre>
<p>We can use the attribute table subsets to select specific polygons
that we want to look at.</p>
<pre class="r"><code># Assign range map of the Eurasian lynx to separate object
lynx_lynx &lt;- mammals[mammals$BINOMIAL==&#39;Lynx lynx&#39;,]

# Map the range
map(&#39;world&#39;)
plot(lynx_lynx, col=&#39;red&#39;, add=T)</code></pre>
<p><img src="2_BiodivData_files/figure-html/unnamed-chunk-6-1.png" width="672" /></p>
<div class="alert alert-success" role="alert">
<p><strong>Test it yourself</strong></p>
<ul>
<li>Select another mammal species and plot the range map.</li>
</ul>
</div>
</div>
</div>
<div id="working-with-range-maps" class="section level1" number="2">
<h1><span class="header-section-number">2</span> Working with range
maps</h1>
<div id="range-size-and-range-centre" class="section level2"
number="2.1">
<h2><span class="header-section-number">2.1</span> Range size and range
centre</h2>
<p>The <code>terra</code> package allows to easily calculate the area of
the polygons, meaning the range size of our species. The function
<code>expanse()</code> outputs the area in square meters, kilometers or
hectars.</p>
<pre class="r"><code># Range area of alpine shrew in square kilometers:
terra::expanse(shrew, unit=&quot;km&quot;)</code></pre>
<pre><code>## [1] 490543.3</code></pre>
<pre class="r"><code># Range area of lynx in square kilometers:
terra::expanse(lynx_lynx, unit=&quot;km&quot;)</code></pre>
<pre><code>##  [1] 2.967561e+03 6.914190e+03 8.054239e+03 3.253875e+03 3.023117e+03
##  [6] 3.845284e+03 3.936121e+03 1.591402e+04 3.451279e+04 1.478952e+03
## [11] 1.459179e+03 8.941187e+02 2.537009e+03 1.146910e+03 1.663037e+03
## [16] 1.614079e+04 1.123943e+03 2.150669e+03 1.998317e+01 4.216219e+02
## [21] 1.443828e+02 1.268629e+02 9.164054e+01 8.175484e+01 7.030462e+01
## [26] 4.550536e+01 5.887205e+01 3.454557e+01 4.305188e+01 4.402397e+01
## [31] 4.070125e+01 4.511146e+01 2.962505e+01 1.322173e+05 1.693141e+07
## [36] 1.310182e+04 5.163677e+03 1.436962e+04 4.393103e+04 2.388271e+03
## [41] 1.835570e+05 4.498608e+04 1.226286e+04 4.414687e+06 1.188083e+04
## [46] 5.946663e+04 2.545997e+00 1.393808e+04</code></pre>
<p>We can also very easily calculate the centre of gravity or range
centroid from the <code>SpatVector</code> object.</p>
<pre class="r"><code># Range centroid:
terra::centroids(shrew)</code></pre>
<pre><code>##  class       : SpatVector 
##  geometry    : points 
##  dimensions  : 1, 27  (geometries, attributes)
##  extent      : 16.02645, 16.02645, 46.90336, 46.90336  (xmin, xmax, ymin, ymax)
##  coord. ref. : lon/lat WGS 84 (EPSG:4326) 
##  names       : id_no      binomial presence origin seasonal compiler yrcompiled
##  type        : &lt;chr&gt;         &lt;chr&gt;    &lt;int&gt;  &lt;int&gt;    &lt;int&gt;    &lt;chr&gt;      &lt;int&gt;
##  values      : 29660 Sorex alpinus        1      1        1     IUCN       2008
##         citation source dist_comm (and 17 more)
##            &lt;chr&gt;  &lt;chr&gt;     &lt;chr&gt;              
##  IUCN (Internat~     NA        NA</code></pre>
<pre class="r"><code># Map the species range and add the centroid to the map
map(&#39;world&#39;,xlim=c(5,30), ylim=c(40,55))
plot(shrew, col=&#39;red&#39;, add=T)
points(terra::centroids(shrew), col=&#39;blue&#39;,cex=3,pch=&quot;+&quot;)</code></pre>
<p><img src="2_BiodivData_files/figure-html/unnamed-chunk-8-1.png" width="672" /></p>
<p>We need to be careful how to interpret these centroids. They
represent the centre of gravity, i.e. the mean coordinates of the
distribution (weighted by cell size) but obviously, if we have several
patches, the centroid might not even fall within an occupied patch.</p>
<div class="alert alert-info">
<p><em><strong>Exercise:</strong></em></p>
<ul>
<li>Use the range map of the mammal species you selected above, and
calculate the range size and range centroid. Add the point location of
your centroid to the range map that you have plotted previously.</li>
</ul>
</div>
</div>
<div id="rasterising-range-maps" class="section level2" number="2.2">
<h2><span class="header-section-number">2.2</span> Rasterising range
maps</h2>
<p>For many applications in ecosystem and biodiversity research, we need
to rasterise the polygons. The problem is that it is unclear at which
spatial resolution the range maps accurately represent species
occurrences. <span class="citation">Hurlbert and Jetz (2007)</span> and
<span class="citation">Jetz, McPherson, and Guralnick (2012)</span>
define the minimum spatial resolution as 100-200km (1-2°), although also
resolutions of 50km (0.5°) and finer have been used <span
class="citation">(Krosby et al. 2015; Zurell et al. 2018)</span>.</p>
<div id="rasterising-range-maps-with-terra" class="section level3"
number="2.2.1">
<h3><span class="header-section-number">2.2.1</span> Rasterising range
maps with <code>terra</code></h3>
<p>Rasterising polygon data is made very easy in the <code>terra</code>
package. We first have to define a <code>SpatRaster</code> object of the
desired resolution, and then transfer the polygon data to the raster
cells.</p>
<pre class="r"><code># By default, terra() will create a 1° resolution map in the *WGS 84* coordinate system (lon/lat).
(r_1deg &lt;- terra::rast())</code></pre>
<pre><code>## class       : SpatRaster 
## dimensions  : 180, 360, 1  (nrow, ncol, nlyr)
## resolution  : 1, 1  (x, y)
## extent      : -180, 180, -90, 90  (xmin, xmax, ymin, ymax)
## coord. ref. : lon/lat WGS 84</code></pre>
<pre class="r"><code># Now, rasterise the shrew polygon data to the 1° raster grid
(shrew_1deg &lt;- terra::rasterize(shrew, r_1deg))</code></pre>
<pre><code>## class       : SpatRaster 
## dimensions  : 180, 360, 1  (nrow, ncol, nlyr)
## resolution  : 1, 1  (x, y)
## extent      : -180, 180, -90, 90  (xmin, xmax, ymin, ymax)
## coord. ref. : lon/lat WGS 84 
## source(s)   : memory
## name        : layer 
## min value   :     1 
## max value   :     1</code></pre>
<pre class="r"><code>map(&#39;world&#39;,xlim=c(5,30), ylim=c(40,55))
plot(shrew, col=&#39;red&#39;, add=T)
plot(shrew_1deg, add=T, alpha=0.6, legend=F)</code></pre>
<p><img src="2_BiodivData_files/figure-html/unnamed-chunk-9-1.png" width="672" /></p>
<p>Obviously, the margins of the range polgyon and the raster map differ
at several places.</p>
<div class="alert alert-success" role="alert">
<p><strong>Test it yourself</strong></p>
<p>Check out the help page <code>?rasterize</code> and find out what the
argument <code>touches</code> is doing. Rasterise the shrew range map
again with setting a different <code>touches</code> argument and map the
result. What is the difference?</p>
</div>
<p>We look at a second example, the lynx:</p>
<pre class="r"><code># Define an empty SpatRaster of the world at 2° spatial resolution
(r_2deg &lt;- terra::rast(res=2))</code></pre>
<pre><code>## class       : SpatRaster 
## dimensions  : 90, 180, 1  (nrow, ncol, nlyr)
## resolution  : 2, 2  (x, y)
## extent      : -180, 180, -90, 90  (xmin, xmax, ymin, ymax)
## coord. ref. : lon/lat WGS 84</code></pre>
<pre class="r"><code># Rasterize the eurasian lynx data
lynx_lynx_2deg &lt;- terra::rasterize(lynx_lynx, r_2deg)

# Map the occupied grid cells
map(&#39;world&#39;)
plot(lynx_lynx_2deg, col=&#39;red&#39;, add = TRUE, legend = FALSE)</code></pre>
<p><img src="2_BiodivData_files/figure-html/unnamed-chunk-10-1.png" width="672" /></p>
</div>
<div id="rasterising-range-maps-with-letsr" class="section level3"
number="2.2.2">
<h3><span class="header-section-number">2.2.2</span> Rasterising range
maps with <code>letsR</code></h3>
<p>There are also specific macroecological packages in R that facilitate
working with range maps and rasterising them, for example the function
<code>lets.presab()</code> in the <code>letsR</code> package.</p>
<pre class="r"><code>library(letsR)

# Let&#39;s look at the Lynx example
# We set the resolution to 1 degree (the default) and restrict the spatial extent to Europe
r_lynx_1deg &lt;- lets.presab(lynx_lynx, resol=1, presence = 1, origin = 1, seasonal = 1)

# Map the range and range centroid
map(&#39;world&#39;)
plot(r_lynx_1deg, add=T, legend=F)</code></pre>
<p><img src="2_BiodivData_files/figure-html/unnamed-chunk-11-1.png" width="672" /></p>
</div>
<div id="bulk-rasterising-multiple-species-range-maps-with-letsr"
class="section level3" number="2.2.3">
<h3><span class="header-section-number">2.2.3</span> Bulk-rasterising
multiple species range maps with <code>letsR</code></h3>
<p>The <code>letsR</code> package also allows to bulk-download multiple
species and rasterise them to form a richness map.</p>
<p>In the <code>lets.presab()</code> function, we can specify which
PRESENCE, ORIGIN and SEASONAL information should be used and which not
(see IUCN metadata on moodle). Let’s look at Neotropical fruit bats
(<strong>Artibeus</strong>) as example. Here, we set
<code>presence=1</code> meaning that we only consider extant species,
<code>origin=1</code> meaning only native species, and
<code>seasonal=1</code> meaning only resident species.</p>
<pre class="r"><code># Subset the SpatVector
artibeus_spp &lt;- mammals[grep(&#39;Artibeus&#39;,mammals$BINOMIAL),]

# Rasterize the ranges using the letsR package
r_artibeus_spp &lt;- lets.presab(artibeus_spp, resol=2, 
                              presence = 1, origin = 1, seasonal = 1)

# Map the species richness
plot(r_artibeus_spp)</code></pre>
<p><img src="2_BiodivData_files/figure-html/unnamed-chunk-12-1.png" width="672" /></p>
<pre class="r"><code># Map single species - here, just the first two
par(mfrow=c(1,2))
plot(r_artibeus_spp, name = &quot;Artibeus amplus&quot;)
plot(r_artibeus_spp, name = &quot;Artibeus anderseni&quot;)</code></pre>
<p><img src="2_BiodivData_files/figure-html/unnamed-chunk-12-2.png" width="672" /></p>
<pre class="r"><code># Look at structure of the object and at the presence-absence matrix
str(r_artibeus_spp, 1)</code></pre>
<pre><code>## List of 3
##  $ Presence_and_Absence_Matrix: num [1:417, 1:22] -109 -107 -109 -107 -105 ...
##   ..- attr(*, &quot;dimnames&quot;)=List of 2
##  $ Richness_Raster            :S4 class &#39;SpatRaster&#39; [package &quot;terra&quot;]
##  $ Species_name               : chr [1:20] &quot;Artibeus amplus&quot; &quot;Artibeus anderseni&quot; &quot;Artibeus aztecus&quot; &quot;Artibeus cinereus&quot; ...
##  - attr(*, &quot;class&quot;)= chr &quot;PresenceAbsence&quot;</code></pre>
<pre class="r"><code>head(r_artibeus_spp$Presence_and_Absence_Matrix)</code></pre>
<pre><code>##      Longitude(x) Latitude(y) Artibeus amplus Artibeus anderseni
## [1,]    -109.1139    26.69894               0                  0
## [2,]    -107.1139    26.69894               0                  0
## [3,]    -109.1139    24.69894               0                  0
## [4,]    -107.1139    24.69894               0                  0
## [5,]    -105.1139    24.69894               0                  0
## [6,]    -101.1139    24.69894               0                  0
##      Artibeus aztecus Artibeus cinereus Artibeus concolor Artibeus fimbriatus
## [1,]                0                 0                 0                   0
## [2,]                0                 0                 0                   0
## [3,]                0                 0                 0                   0
## [4,]                1                 0                 0                   0
## [5,]                1                 0                 0                   0
## [6,]                1                 0                 0                   0
##      Artibeus fraterculus Artibeus glaucus Artibeus gnomus Artibeus hirsutus
## [1,]                    0                0               0                 1
## [2,]                    0                0               0                 1
## [3,]                    0                0               0                 1
## [4,]                    0                0               0                 1
## [5,]                    0                0               0                 1
## [6,]                    0                0               0                 0
##      Artibeus incomitatus Artibeus inopinatus Artibeus jamaicensis
## [1,]                    0                   0                    0
## [2,]                    0                   0                    0
## [3,]                    0                   0                    0
## [4,]                    0                   0                    1
## [5,]                    0                   0                    1
## [6,]                    0                   0                    0
##      Artibeus lituratus Artibeus obscurus Artibeus phaeotis
## [1,]                  1                 0                 0
## [2,]                  1                 0                 0
## [3,]                  1                 0                 0
## [4,]                  1                 0                 1
## [5,]                  1                 0                 1
## [6,]                  0                 0                 0
##      Artibeus planirostris Artibeus rosenbergii Artibeus toltecus
## [1,]                     0                    0                 1
## [2,]                     0                    0                 1
## [3,]                     0                    0                 1
## [4,]                     0                    0                 1
## [5,]                     0                    0                 1
## [6,]                     0                    0                 1
##      Artibeus watsoni
## [1,]                0
## [2,]                0
## [3,]                0
## [4,]                0
## [5,]                0
## [6,]                0</code></pre>
<div class="alert alert-info">
<p><em><strong>Exercise:</strong></em></p>
<ul>
<li>Select another group of mammals, and follow the workflow to
rasterise ranges and map species richness of the species within that
genus.</li>
<li>Download climate data at the correct resolution, and extract the
climate information for your species richness data. Use a scatter plot
to display the species richness against mean annual temperature.</li>
</ul>
</div>
</div>
</div>
</div>
<div id="red-list-data" class="section level1" number="3">
<h1><span class="header-section-number">3</span> Red list data</h1>
<p>The IUCN Red List of threatened species is a useful resource when
analysing diversity changes, species range changes and population
changes: www.iucnredlist.org <span class="citation">(IUCN 2019)</span>.
It provides information on the global conservation status of species,
specifically of animals, fungi and plant species. The IUCN has defined a
set of criteria to evaluate the extinction risk of species. According to
these criteria, species are classified into nine different categories
although strictly only five categories describe the conservation status
- from least concern to critically endangered [Fig. 1; <span
class="citation">IUCN (2012)</span>].</p>
<div class="figure">
<img src="figures/redlist.png" alt="**Figure 1. The IUCN red list categories. Adapted from [@IUCN2012].**" width="70%" />
<p class="caption">
<strong>Figure 1. The IUCN red list categories. Adapted from <span
class="citation">(IUCN 2012)</span>.</strong>
</p>
</div>
<p>The IUCN distinguishes five criteria that are used to classify
species into one of the three threatened categories (Fig. 1; <span
class="citation">IUCN (2012)</span>):</p>
<ul>
<li>A. Population size reduction</li>
<li>B. Extent of occurrence (B1) or Area of occupancy (B2)</li>
<li>C. Small population size and decline</li>
<li>D. Very small or restricted population</li>
<li>E. Quantitative analysis (probability of extinction within next 100
years)</li>
</ul>
<p>All of this information (the red list category, the relevant
assessment criteria, the population trends, etc.) are provided by the
IUCN. We can look at all these information online, e.g. for the <a
href="https://www.iucnredlist.org/species/12519/121707666">Eurasian
lynx</a>.</p>
<p>Of course, if you want to analyse your species data in light of these
IUCN assessments, it can become very tedious to look up all information
by hand. The IUCN red list team provides an API for this purpose, and
the R package <code>rredlist</code> is a client to access this <a
href="http://apiv3.iucnredlist.org/api/v3/docs">IUCN red list API</a>.
It requires an <a href="http://apiv3.iucnredlist.org/api/v3/docs">API
key / a token</a> to use the API. You have to indicate your research
objectives for using the API.</p>
<div id="mammal-data" class="section level2" number="3.1">
<h2><span class="header-section-number">3.1</span> Mammal data</h2>
<p>We will work with European mammals. The distribution data come from
the IUCN range maps above that we rasterised to European extent. Course
participants can download the data from the moodle page (store in your
<em>data</em> folder). For completeness, I show the code below for
compiling the data.</p>
<pre class="r"><code># rough European extent
extent_eur &lt;- c(-15,45,35,71)

# Crop SpatRaster to European extent
mammals_eur &lt;- terra::crop(mammals, extent_eur)

# Convert to old raster format:
mammals_eur.sp &lt;- as(mammals_eur, &quot;Spatial&quot;)

# Rasterise mammal range maps for European extent
r_mammals_eur &lt;- lets.presab(mammals_eur.sp, resol=1, presence = 1, origin = 1, seasonal = 1)

# Save the PresenceAbsence object
save(extent_eur, r_mammals_eur,file=&#39;data/r_mammals_eur_1deg.RData&#39;)</code></pre>
<p>Load the <code>PresenceAbsence</code> object:</p>
<pre class="r"><code>load(&#39;data/r_mammals_eur_1deg.RData&#39;)

# Remember how the PresenceAbsence objects look like:
str(r_mammals_eur,2)</code></pre>
<pre><code>## List of 3
##  $ Presence_and_Absence_Matrix: num [1:1491, 1:293] -14.5 -13.5 -12.5 -11.5 -10.5 -9.5 -8.5 -7.5 -6.5 -5.5 ...
##   ..- attr(*, &quot;dimnames&quot;)=List of 2
##  $ Richness_Raster            :Formal class &#39;RasterLayer&#39; [package &quot;raster&quot;] with 13 slots
##  $ Species_name               : chr [1:291] &quot;Acomys cilicicus&quot; &quot;Acomys minous&quot; &quot;Alces alces&quot; &quot;Allactaga elater&quot; ...
##  - attr(*, &quot;class&quot;)= chr &quot;PresenceAbsence&quot;</code></pre>
<pre class="r"><code># Plot the richness raster:
plot(raster::crop(r_mammals_eur$Richness_Raster,extent_eur))</code></pre>
<p><img src="2_BiodivData_files/figure-html/unnamed-chunk-15-1.png" width="672" /></p>
<p>For merging the mammal distribution with the red list data, we will
need the species names:</p>
<pre class="r"><code># Extract the species names from the PresenceAbsence object
names_mammals_eur &lt;- colnames(r_mammals_eur$Presence_and_Absence_Matrix)[-c(1:2)]</code></pre>
</div>
<div id="iucn-red-list-data" class="section level2" number="3.2">
<h2><span class="header-section-number">3.2</span> IUCN Red list
data</h2>
<p>To access the red list through R (via an API) you need an API token.
Course participants can download an example token to execute the
following code or the resulting data
(<em>mammals_eur_redlist_status.csv</em> and
<em>mammals_eur_redlist_threats.csv</em>) from the secured moodle folder
(but please be aware that the <a
href="https://www.iucnredlist.org/terms/terms-of-use">IUCN terms of
use</a> apply!). External readers are advised to generate an API
token:</p>
<pre class="r"><code>library(rredlist)

# Generate your personal API token
rl_use_iucn()</code></pre>
<div id="iucn-red-list-status" class="section level3" number="3.2.1">
<h3><span class="header-section-number">3.2.1</span> IUCN red list
status</h3>
<p>If you have your own IUCN API key, you can easily download the
information on species’ red list categories:</p>
<pre class="r"><code># Download red list category for single species using your personal API token &quot;MY_IUCN_REDLIST_KEY&quot;
(rl_search(&#39;Lynx lynx&#39;, key= MY_IUCN_REDLIST_KEY))</code></pre>
<pre><code>## $name
## [1] &quot;Lynx lynx&quot;
## 
## $result
##   taxonid scientific_name  kingdom   phylum    class     order  family genus
## 1   12519       Lynx lynx ANIMALIA CHORDATA MAMMALIA CARNIVORA FELIDAE  Lynx
##   main_common_name        authority published_year assessment_date category
## 1    Eurasian Lynx (Linnaeus, 1758)           2015      2014-04-20       LC
##   criteria population_trend marine_system freshwater_system terrestrial_system
## 1       NA           Stable         FALSE             FALSE               TRUE
##                                                                                                 assessor
## 1 Breitenmoser, U., Breitenmoser-Würsten, C., Lanz, T., von Arx, M., Antonevich, A., Bao, W. &amp; Avgan, B.
##                              reviewer aoo_km2  eoo_km2 elevation_upper
## 1 Nowell, K., Hunter, L. &amp; Mallon, D.      NA 42765298            5500
##   elevation_lower depth_upper depth_lower errata_flag
## 1               0          NA          NA        TRUE
##                                                                                                                               errata_reason
## 1 An errata version of the original assessment has been published to correct the typo &quot;Mmountain Hares&quot; which should read &quot;Mountain Hares&quot;.
##   amended_flag amended_reason
## 1           NA             NA</code></pre>
<pre class="r"><code># Download red list categories for all species
redlist_status &lt;- do.call(rbind,lapply(names_mammals_eur,FUN=function(sp){
  rl_search(sp, key= MY_IUCN_REDLIST_KEY)$result
  }
  ))</code></pre>
<p>Course participants can download the data from moodle and read it
in:</p>
<pre class="r"><code>redlist_status &lt;- read.table(&#39;data/mammals_eur_redlist_status.csv&#39;, header=T, sep=&#39;;&#39;)</code></pre>
<p>Here is an example of the kind of information in the table. Compare
this to the information given on the IUCN website, e.g. for the <a
href="https://www.iucnredlist.org/species/12519/121707666">Eurasian
Lynx</a>.</p>
<pre class="r"><code>redlist_status[1:10,10:20]</code></pre>
<pre><code>##               authority published_year assessment_date category criteria
## 1   Spitzenberger, 1978           2019      2017-05-22       DD     &lt;NA&gt;
## 2            Bate, 1906           2019      2017-05-22       DD     &lt;NA&gt;
## 3      (Linnaeus, 1758)           2016      2015-09-27       LC     &lt;NA&gt;
## 4  (Lichtenstein, 1825)           2016      2016-09-08       LC     &lt;NA&gt;
## 5          Thomas, 1881           2019      2019-07-09       LC     &lt;NA&gt;
## 6          (Kerr, 1792)           2016      2016-09-08       LC     &lt;NA&gt;
## 7          Thomas, 1897           2016      2016-09-09       LC     &lt;NA&gt;
## 8        (Pallas, 1777)           2021      2020-08-26       VU       C1
## 9        (Pallas, 1771)           2016      2016-09-08       LC     &lt;NA&gt;
## 10       Heinrich, 1952           2016      2016-09-09       LC     &lt;NA&gt;
##    population_trend marine_system freshwater_system terrestrial_system
## 1           Unknown         FALSE             FALSE               TRUE
## 2           Unknown         FALSE             FALSE               TRUE
## 3        Increasing         FALSE             FALSE               TRUE
## 4        Decreasing         FALSE             FALSE               TRUE
## 5        Decreasing         FALSE             FALSE               TRUE
## 6        Decreasing         FALSE             FALSE               TRUE
## 7        Decreasing         FALSE             FALSE               TRUE
## 8        Decreasing         FALSE             FALSE               TRUE
## 9            Stable         FALSE             FALSE               TRUE
## 10          Unknown         FALSE             FALSE               TRUE
##                                                                                                                  assessor
## 1                                                                                              Gerrie, R. &amp; Kennerley, R.
## 2                                                                                              Gerrie, R. &amp; Kennerley, R.
## 3                                                                                                         Hundertmark, K.
## 4                Shenbrot, G., Tsytsulina, K., Batsaikhan, N., Avirmed, D., Tinnin, D., Sukhchuluun, G &amp; Lkhagvasuren, D.
## 5                                                                                               Dando, T. &amp; Kennerley, R.
## 6                                                             Tsytsulina, K., Formozov, N., Zagorodnyuk, I. &amp; Sheftel, B.
## 7                                                                                      Eken, G., Bozdogan, M. &amp; Molur, S.
## 8  Cassinello, J., Bounaceur, F., Brito, J.C., Bussière, E., Cuzin, F., Gil-Sánchez, J., Herrera-Sánchez, F. &amp; Wacher, T.
## 9                Kaneko, Y., Kryštufek, B., Zagarondnyuk, I., Vohralík, V., Batsaikhan, N., Avirmed, D. &amp; Sukhchuluun, G.
## 10                                                                                                             Meinig, H.
##                      reviewer
## 1                   Amori, G.
## 2                   Amori, G.
## 3  Brook, S.M. &amp; McShea, W.J.
## 4                   Amori, G.
## 5                   Amori, G.
## 6                   Amori, G.
## 7                   Amori, G.
## 8    Herrero, J. &amp; Michel, S.
## 9                   Amori, G.
## 10                  Amori, G.</code></pre>
<p>We can now look at how many mammal species are listed in the
different red list categories or how many species have declining, stable
or increasing population trends:</p>
<pre class="r"><code># Conservation status
table(redlist_status$category)</code></pre>
<pre><code>## 
##  CR  DD  EN  LC  NT  VU 
##   7  12  10 213  18  21</code></pre>
<pre class="r"><code># Population trends
table(redlist_status$population_trend)</code></pre>
<pre><code>## 
## Decreasing Increasing     Stable    Unknown 
##         76          8        102         95</code></pre>
<p>We can also extract all species that are classified as
vulnerable:</p>
<pre class="r"><code>(subset(redlist_status,category==&#39;VU&#39;)$scientific_name)</code></pre>
<pre><code>##  [1] &quot;Ammotragus lervia&quot;        &quot;Arvicola sapidus&quot;        
##  [3] &quot;Dinaromys bogdanovi&quot;      &quot;Gazella cuvieri&quot;         
##  [5] &quot;Gazella dorcas&quot;           &quot;Gazella subgutturosa&quot;    
##  [7] &quot;Lepus castroviejoi&quot;       &quot;Lepus corsicanus&quot;        
##  [9] &quot;Mesocricetus auratus&quot;     &quot;Miniopterus schreibersii&quot;
## [11] &quot;Myomimus roachi&quot;          &quot;Myotis capaccinii&quot;       
## [13] &quot;Nyctalus lasiopterus&quot;     &quot;Panthera pardus&quot;         
## [15] &quot;Plecotus sardus&quot;          &quot;Rangifer tarandus&quot;       
## [17] &quot;Rhinolophus mehelyi&quot;      &quot;Spalax graecus&quot;          
## [19] &quot;Spalax zemni&quot;             &quot;Ursus maritimus&quot;         
## [21] &quot;Vormela peregusna&quot;</code></pre>
<div class="alert alert-info">
<p><strong>Exercise:</strong></p>
<p>Explore the red list status table. For example,</p>
<ul>
<li>Identify all species classified as endangered or critically
endangered.</li>
<li>What is the red list status of the species with decreasing
population trend?</li>
</ul>
</div>
</div>
<div id="iucn-red-list-threats" class="section level3" number="3.2.2">
<h3><span class="header-section-number">3.2.2</span> IUCN red list
threats</h3>
<p>The IUCN also assesses the main threats per species as you saw in the
<a href="https://www.iucnredlist.org/species/12519/121707666">Eurasian
lynx</a> example. The Threat Classification Scheme can be found <a
href="https://www.iucnredlist.org/resources/threat-classification-scheme">here</a>.</p>
<p>If you have your own IUCN API key, you can easily download the
information on species’ red list threats:</p>
<pre class="r"><code># Download red list threats for single species
rl_threats(&#39;Lynx lynx&#39;, key= MY_IUCN_REDLIST_KEY)</code></pre>
<pre><code>## $name
## [1] &quot;Lynx lynx&quot;
## 
## $result
##     code                                                         title  timing
## 1    2.1                           Annual &amp; perennial non-timber crops Ongoing
## 2  2.1.2                                          Small-holder farming Ongoing
## 3  2.1.3                                         Agro-industry farming Ongoing
## 4    2.2                                       Wood &amp; pulp plantations Ongoing
## 5  2.2.1                                      Small-holder plantations Ongoing
## 6  2.2.2                                     Agro-industry plantations Ongoing
## 7    2.3                                  Livestock farming &amp; ranching Ongoing
## 8  2.3.1                                               Nomadic grazing Ongoing
## 9  2.3.2                     Small-holder grazing, ranching or farming Ongoing
## 10 2.3.3                    Agro-industry grazing, ranching or farming Ongoing
## 11   3.2                                            Mining &amp; quarrying Ongoing
## 12   4.1                                             Roads &amp; railroads Ongoing
## 13   5.1                        Hunting &amp; trapping terrestrial animals Ongoing
## 14 5.1.1                       Intentional use (species is the target) Ongoing
## 15 5.1.2             Unintentional effects (species is not the target) Ongoing
## 16 5.1.3                                           Persecution/control Ongoing
## 17   5.3                                     Logging &amp; wood harvesting Ongoing
## 18 5.3.5                                 Motivation Unknown/Unrecorded Ongoing
## 19   8.1 Invasive non-native/alien species/diseases (Canis familiaris) Ongoing
## 20 8.1.2                              Named species (Canis familiaris) Ongoing
##              scope            severity         score         invasive
## 1             &lt;NA&gt;                &lt;NA&gt; Low Impact: 3             &lt;NA&gt;
## 2             &lt;NA&gt;                &lt;NA&gt; Low Impact: 3             &lt;NA&gt;
## 3             &lt;NA&gt;                &lt;NA&gt; Low Impact: 3             &lt;NA&gt;
## 4             &lt;NA&gt;                &lt;NA&gt; Low Impact: 3             &lt;NA&gt;
## 5             &lt;NA&gt;                &lt;NA&gt; Low Impact: 3             &lt;NA&gt;
## 6             &lt;NA&gt;                &lt;NA&gt; Low Impact: 3             &lt;NA&gt;
## 7             &lt;NA&gt;                &lt;NA&gt; Low Impact: 3             &lt;NA&gt;
## 8             &lt;NA&gt;                &lt;NA&gt; Low Impact: 3             &lt;NA&gt;
## 9             &lt;NA&gt;                &lt;NA&gt; Low Impact: 3             &lt;NA&gt;
## 10            &lt;NA&gt;                &lt;NA&gt; Low Impact: 3             &lt;NA&gt;
## 11            &lt;NA&gt;                &lt;NA&gt; Low Impact: 3             &lt;NA&gt;
## 12            &lt;NA&gt;                &lt;NA&gt; Low Impact: 3             &lt;NA&gt;
## 13            &lt;NA&gt;                &lt;NA&gt; Low Impact: 3             &lt;NA&gt;
## 14            &lt;NA&gt;                &lt;NA&gt; Low Impact: 3             &lt;NA&gt;
## 15            &lt;NA&gt;                &lt;NA&gt; Low Impact: 3             &lt;NA&gt;
## 16            &lt;NA&gt;                &lt;NA&gt; Low Impact: 3             &lt;NA&gt;
## 17            &lt;NA&gt;                &lt;NA&gt; Low Impact: 3             &lt;NA&gt;
## 18            &lt;NA&gt;                &lt;NA&gt; Low Impact: 3             &lt;NA&gt;
## 19 Minority (&lt;50%) Negligible declines Low Impact: 4 Canis familiaris
## 20 Minority (&lt;50%) Negligible declines Low Impact: 4 Canis familiaris</code></pre>
<pre class="r"><code># Download red list threats for all species
redlist_threats &lt;- do.call(rbind,
                           lapply(seq_len(length(names_mammals_eur)),FUN=function(i){
                             xi &lt;- rl_threats(names_mammals_eur[i], key= MY_IUCN_REDLIST_KEY); 
                             if(length(xi$result)) {
                               data.frame(species=names_mammals_eur[i],xi$result) 
                               }
                             }
                             ))</code></pre>
<p>Course participants can download the data from moodle and read it
in:</p>
<pre class="r"><code>redlist_threats &lt;- read.table(&#39;data/mammals_eur_redlist_threats.csv&#39;, header=T, sep=&#39;;&#39;)</code></pre>
<p>The threats are ordered hierarchically from broad threat type to very
detailed threat, e.g.:</p>
<ul>
<li><em>2 Agriculture &amp; aquaculture</em> &gt; <em>2.2 Wood &amp;
pulp plantations </em> &gt; <em>2.2.1 Small-holder plantations</em></li>
<li><em>5 Biological resource use</em> &gt; <em>5.4 Fishing &amp;
harvesting aquatic resources</em> &gt; <em>5.4.3 Unintentional effects:
subsistence/small scale (species being assessed is not the
target)[harvest]</em></li>
</ul>
<p>Here is an example of the kind of information in the table. For more
details, please have a look at the IUCN website, e.g. the <a
href="https://www.iucnredlist.org/species/12519/121707666">Eurasian
lynx</a> example, and at the <a
href="https://www.iucnredlist.org/resources/threat-classification-scheme">Threat
Classification Scheme</a>.</p>
<pre class="r"><code>redlist_threats[sample(nrow(redlist_threats),10),-c(1:2)]</code></pre>
<pre><code>##                                                          title  timing
## 139                     Hunting &amp; trapping terrestrial animals Ongoing
## 1055                    Hunting &amp; trapping terrestrial animals Ongoing
## 947  Problematic native species/diseases (Capreolus capreolus) Ongoing
## 520                    Intentional use (species is the target) Ongoing
## 586                                    Wood &amp; pulp plantations Ongoing
## 163                    Intentional use (species is the target) Ongoing
## 148                    Intentional use (species is the target) Ongoing
## 958                                                   Droughts Ongoing
## 32                         Annual &amp; perennial non-timber crops Ongoing
## 712                          Agricultural &amp; forestry effluents Ongoing
##                  scope            severity         score            invasive
## 139               &lt;NA&gt;                &lt;NA&gt; Low Impact: 3                &lt;NA&gt;
## 1055   Minority (&lt;50%)                &lt;NA&gt; Low Impact: 4                &lt;NA&gt;
## 947               &lt;NA&gt;                &lt;NA&gt; Low Impact: 3 Capreolus capreolus
## 520  Majority (50-90%) Negligible declines Low Impact: 5                &lt;NA&gt;
## 586               &lt;NA&gt;                &lt;NA&gt;          &lt;NA&gt;                &lt;NA&gt;
## 163               &lt;NA&gt;                &lt;NA&gt; Low Impact: 3                &lt;NA&gt;
## 148               &lt;NA&gt;                &lt;NA&gt; Low Impact: 3                &lt;NA&gt;
## 958    Minority (&lt;50%)             Unknown       Unknown                &lt;NA&gt;
## 32                &lt;NA&gt;                &lt;NA&gt; Low Impact: 3                &lt;NA&gt;
## 712               &lt;NA&gt;                &lt;NA&gt; Low Impact: 3                &lt;NA&gt;</code></pre>
<p>We can extract many useful information from this table, for example,
when specific threats occurred.</p>
<pre class="r"><code>table(redlist_threats$timing)</code></pre>
<pre><code>## 
##                   Future                  Ongoing Past, Unlikely to Return 
##                       35                     1106                       28</code></pre>
</div>
</div>
</div>
<div id="homework-prep" class="section level1" number="4">
<h1><span class="header-section-number">4</span> Homework prep</h1>
<p>For the homework, you will need several objects that you should not
forget to save.</p>
<pre class="r"><code># Write terra objects to file:
terra::writeVector(lynx_lynx, &#39;data/lynx_lynx.shp&#39;) # exchange with the mammal species you chose to work with

# Save other (non-terra) objects from the workspace:
save(redlist_status, redlist_threats, file=&#39;data/2_BiodivData.RData&#39;)</code></pre>
<p>As homework, solve all the exercises in the blue boxes.</p>
<div class="alert alert-info">
<p><strong>Exercise:</strong></p>
<p>Explore the threats table. For example,</p>
<ul>
<li>Pick a species and identify which threats are causing rapid declines
and slow declines.</li>
</ul>
</div>
</div>
<div id="references" class="section level1 unnumbered">
<h1 class="unnumbered">References</h1>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-Hurlbert2007" class="csl-entry">
Hurlbert, A. H., and W. Jetz. 2007. <span>“Species Richness, Hotspots,
and the Scale Dependence of Range Maps in Ecology and
Conservation.”</span> <em>PNAS</em> 104: 13384–89.
</div>
<div id="ref-IUCN2012" class="csl-entry">
IUCN. 2012. <em>IUCN Red List Categories and Criteria: Version 3.1.</em>
Gland, Switzerland; Cambridge, UK: IUCN.
</div>
<div id="ref-IUCN2019" class="csl-entry">
———. 2019. <em>The <span>IUCN</span> Red List of Threatened Species.
Version 2019-2.</em> http://www.iucnredlist.org. Downloaded on 27
October 2019.
</div>
<div id="ref-Jetz2012a" class="csl-entry">
Jetz, W., J. M. McPherson, and R. P. Guralnick. 2012. <span>“Integrating
Biodiversity Distribution Knowledge: Toward a Global Map of
Life.”</span> <em>Trends in Ecology &amp; Evolution</em> 27: 151–59.
</div>
<div id="ref-Krosby2015" class="csl-entry">
Krosby, M., C. B. Wilsey, J. L. McGuire, J. M. Duggan, T. M. Nogeire, J.
A. Heinrichs, J. J. Tewksbury, and J. J. Lawler. 2015.
<span>“Climate-Induced Range Overlap Among Closely Related
Species.”</span> <em>Nature Climate Change</em> 5: 883–86.
</div>
<div id="ref-W_est_2020" class="csl-entry">
Wüest, R. O., N. E. Zimmermann, D. Zurell, J. M. Alexander, S. A. Fritz,
C. Hof, H. Kreft, et al. 2020. <span>“Macroecology in the Age of Big
Data <span></span> Where to Go from Here?”</span> <em>Journal of
Biogeography</em> 47 (1): 1–12. <a
href="https://doi.org/10.1111/jbi.13633">https://doi.org/10.1111/jbi.13633</a>.
</div>
<div id="ref-Zurell2018c" class="csl-entry">
Zurell, D., C. H. Graham, L. Gallien, W. Thuiller, and N. E. Zimmermann.
2018. <span>“Long-Distance Migratory Birds Threatened by Multiple
Independent Risks from Global Change.”</span> <em>Nature Climate
Change</em> 8 (October): 992–96. <a
href="https://doi.org/10.1038/s41558-018-0312-9">https://doi.org/10.1038/s41558-018-0312-9</a>.
</div>
</div>
</div>

<!DOCTYPE html>
<html>

<br>
<hr />
<div id="footer">
<p>Damaris Zurell 2023 <a href="http://creativecommons.org/licenses/by/4.0/" >(CC BY 4.0)</a>.  </p>
</div>

</html>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
