<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Getting started with RangeShiftR</title>

<script src="site_libs/header-attrs-2.25/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/clipboard-1.7.1/clipboard.min.js"></script>
<link href="site_libs/primer-tooltips-1.4.0/build.css" rel="stylesheet" />
<link href="site_libs/klippy-0.0.0.9500/css/klippy.min.css" rel="stylesheet" />
<script src="site_libs/klippy-0.0.0.9500/js/klippy.min.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>





<style type="text/css">
/* for pandoc --citeproc since 2.11 */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Ecosystem dynamics and biodiversity</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Pracs: Spatial data
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="0_setup.html">0. Getting started</a>
    </li>
    <li>
      <a href="1_SpatialData.html">1. Spatial data in R</a>
    </li>
    <li>
      <a href="2_BiodivData.html">2. Biodiversity data in R</a>
    </li>
    <li>
      <a href="3_EnvData.html">3. Environmental data in R</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Pracs: Species distribution models
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="4_SDM_intro.html">4. SDMs: simple model fitting</a>
    </li>
    <li>
      <a href="5_SDM_eval.html">5. SDMs: assessment and prediction</a>
    </li>
    <li>
      <a href="6_SDM_algorithms.html">6. SDMs: algorithms and ensembles</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Pracs: population modelling
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="7_RS_intro.html">7. Getting started with RangeShiftR</a>
    </li>
    <li>
      <a href="8_RS_lynx.html">8. RangeShiftR: Eurasian lynx reintroduction</a>
    </li>
    <li>
      <a href="9_RS_grouse.html">9. RangeShiftR: Black grouse range dynamics</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Pracs: ecosystem modelling
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="10_Mad_intro.html">10. Running MadingleyR</a>
    </li>
    <li>
      <a href="11_Mad_carnivores.html">11. MadingleyR: Role of large carnivores</a>
    </li>
    <li>
      <a href="12_Mad_landuse.html">12. MadingleyR: land use change effects</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://www.uni-potsdam.de/en/ibb-macroecology/index">
    <span class="fa fa-home"></span>
     
  </a>
</li>
<li>
  <a href="https://twitter.com/ZurellLab">
    <span class="fa fa-twitter"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Getting started with RangeShiftR</h1>

</div>


<script>
  addClassKlippyTo("pre.r, pre.markdown");
  addKlippy('right', 'top', 'auto', '1', 'Copy code', 'Copied!');
</script>
<div class="alert alert-info">
<p><strong>RStudio project</strong></p>
<p>Open the RStudio project that we created in the first session. I
recommend to use this RStudio project for the entire course and within
the RStudio project create separate R scripts for each session.</p>
<ul>
<li>Create a new empty R script by going to the tab “File”, select “New
File” and then “R script”</li>
<li>In the new R script, type
<code># Session 7: Getting started with RangeShiftR</code> and save the
file in your folder “scripts” within your project folder, e.g. as
“7_RS_intro.R”</li>
</ul>
</div>
<p><img src="figures/RangeShiftR_Hexsticker.png" width="20%" /></p>
<p>In the next three practicals, we are going to work with the
individual-based modelling platform <em>RangeShifter</em>, which
simulates population dynamics and dispersal behaviour on spatially
explicit landscapes <span class="citation">(G. Bocedi et al. 2014; Greta
Bocedi et al. 2021)</span>. The platform provides functionality for a
wide variety of modelling applications ranging from applied questions,
where it can be parameterised for real landscapes and species to compare
alternative potential management interventions, to purely theoretical
studies of species’ eco-evolutionary dynamics and responses to different
environmental pressures.</p>
<p><img src="figures/RS_platform.png" width="70%" /></p>
<p>There are different software options available including a Windows
GUI and a command line version <span class="citation">(Greta Bocedi et
al. 2021)</span> as well as an R package <span class="citation">(Malchow
et al. 2021)</span>. More information on the full functionality and
software can be found on the website <a
href="https://rangeshifter.github.io"
class="uri">https://rangeshifter.github.io</a>. Here, we are going to
use the R package <code>RangeShiftR</code> and concentrate on the
processes demography and dispersal (and ignore evolution). Should you
wish to explore the R package in more depth than this course can cover,
please have a look at our more elaborate <a
href="https://rangeshifter.github.io/RangeshiftR-tutorials/">software
tutorials</a>.</p>
<p>The <code>RangeShiftR</code> package source is publicly released on a
<a
href="https://github.com/RangeShifter/RangeShiftR-package"><em>github</em>
repository</a> as a beta version (it is not available on
<em>CRAN</em>).</p>
<div class="alert alert-danger">
<p><strong>INSTALLING RangeShiftR</strong></p>
<p>Please follow the installation guide here: <a
href="https://rangeshifter.github.io/RangeshiftR-tutorials/installing.html"
class="uri">https://rangeshifter.github.io/RangeshiftR-tutorials/installing.html</a></p>
</div>
<div id="introduction-to-rangeshiftr" class="section level1" number="1">
<h1><span class="header-section-number">1</span> Introduction to
RangeShiftR</h1>
<p>In this overview, we quickly go through the main functionality of the
<em>RangeShiftR</em> R-package in order to get a first overview of the
basic structure. It is also meant for setting up the appropriate folder
structures on your machine.</p>
<div id="rangeshiftr-workflow" class="section level2" number="1.1">
<h2><span class="header-section-number">1.1</span>
<code>RangeShiftR</code> workflow</h2>
<p>Within the standard workflow of <em>RangeShiftR</em>, a simulation is
defined by:</p>
<ol style="list-style-type: decimal">
<li>a so-called parameter master object that contains the simulation
modules, which represent the model structure, as well as the (numeric)
values of all necessary simulation parameters, and</li>
<li>the path to the working directory on the disc where the simulation
inputs and outputs are stored.</li>
</ol>
<p>The standard workflow of <em>RangeShiftR</em> is to load input maps
from ASCII raster files and to write all simulation output into text
files. Therefore, the specified working directory needs to have a
certain folder structure: It should contain 3 sub-folders named
‘Inputs’, ‘Outputs’ and ‘Output_Maps’.</p>
</div>
<div id="running-a-first-working-example" class="section level2"
number="1.2">
<h2><span class="header-section-number">1.2</span> Running a first
working example</h2>
<p>Load the package by typing:</p>
<pre class="r"><code>library(RangeShiftR)</code></pre>
<pre><code>## RangeshiftR version 1.0.4 (10.05.2022)
## Copyright (C) 2020-2022 Anne-Kathleen Malchow, Greta Bocedi, Stephen C.F. Palmer, Justin M.J. Travis, Damaris Zurell
## 
## This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY.
## You are welcome to redistribute it and/or modify it under certain conditions;
## type &#39;RangeShiftR_license()&#39; for details.</code></pre>
<p>Create a parameter master object with all the default settings and
store it:</p>
<pre class="r"><code>s &lt;- RSsim()</code></pre>
<p>Use your file explorer on your machine, navigate to the “models”
folder within your project, and create a sub-folder for the current
practical called “RS_GettingStarted”. Next, we go back to your RStudio
project and store the path in a variable. This can either be the
relative path from your R working directory or the absolute path.</p>
<pre class="r"><code>dirpath = &quot;models/RS_GettingStarted/&quot;</code></pre>
<p>Create the RS folder structure, if it doesn’t yet exist:</p>
<pre class="r"><code>dir.create(paste0(dirpath,&quot;Inputs&quot;), showWarnings = TRUE)
dir.create(paste0(dirpath,&quot;Outputs&quot;), showWarnings = TRUE)
dir.create(paste0(dirpath,&quot;Output_Maps&quot;), showWarnings = TRUE)</code></pre>
<p>With this, we are already set to run our first simulation by
typing:</p>
<pre class="r"><code>RunRS(s,dirpath)</code></pre>
<pre><code>## Checking Control parameters 
## 
## Control Parameters checked
## 
## Run Simulation(s) with random seed ...
## 
## Running simulation nr. 1
## 
## starting replicate 0
## RunModel(): completed initialisation 
## starting year 0...
## starting year 1...
## starting year 2...
## starting year 3...
## starting year 10...
## starting year 20...
## starting year 30...
## 
## starting replicate 1
## RunModel(): completed initialisation 
## starting year 0...
## starting year 1...
## starting year 2...
## starting year 3...
## starting year 10...
## starting year 20...
## starting year 30...
## 
## ***** Elapsed time: 2 seconds
## 
## *****
## ***** Simulation completed 
## ***** Outputs folder: /Users/zurell/data/Lehre/UP_Lehre/EEC/ConsBiogeogr/qcb-course/models/GettingStarted/Outputs/
## *****</code></pre>
<pre><code>## $Errors
## [1] 0</code></pre>
<p>You should find the generated output - the simulation results as well
as some log files - in the ‘Outputs’ folder.</p>
</div>
</div>
<div id="simulation-modules" class="section level1" number="2">
<h1><span class="header-section-number">2</span> Simulation modules</h1>
<p>To look at the parameter master in more detail, simply type:</p>
<pre class="r"><code>s</code></pre>
<pre><code>##  Batch # 1 
## 
##  Simulation # 1 
##  -----------------
##    Replicates =  2 
##    Years      =  50 
##    Absorbing  =  FALSE 
##  File Outputs:
##    Range,       every 1 years
##    Populations, every 1 years, starting year 0
## 
##  Artificial landscape: random structure, binary habitat/matrix code
##    Size         : 65 x 65 cells
##    Resolution   :       100 meters
##    Proportion of suitable habitat: 0.5 
##    K or 1/b     : 10 
## 
##  Demography:
##   Unstructured population:
##    Rmax : 1.5 
##    bc   : 1 
##   Reproduction Type : 0 (female only)
## 
##  Dispersal: 
##   Emigration:
##    Emigration probabilities:
##      [,1]
## [1,]    0
## 
##   Transfer:
##    Dispersal Kernel
##    Dispersal kernel traits:
##      [,1]
## [1,]  100
##    Constant mortality prob = 0 
## 
##   Settlement:
##    Settlement conditions:
##      [,1]
## [1,]    0
##    FindMate = FALSE 
## 
##  Initialisation: 
##    InitType = 0 : Free initialisation 
##                   of all suitable cells/patches.
##    InitDens = 1 : At half K_or_DensDep</code></pre>
<p>It contains of a number of parameter modules that each define
different aspects of the RangeShifter simulation. Specifically, there
are:</p>
<ul>
<li>Simulation</li>
<li>Landscape</li>
<li>Demography</li>
<li>Dispersal</li>
<li>Genetics</li>
<li>Initialisation</li>
</ul>
<p>Here is a schematic overview of the module constructors and their
relations <span class="citation">(Malchow et al. 2021)</span>:</p>
<div class="figure">
<img src="figures/func_overview.png" alt="*RangeshiftR function overview*" width="1604" />
<p class="caption">
<em>RangeshiftR function overview</em>
</p>
</div>
<p>In the following, we go through some of the most relevant aspects of
each module.</p>
<div id="simulation" class="section level2" number="2.1">
<h2><span class="header-section-number">2.1</span> Simulation</h2>
<p>This module is used to set general simulation parameters
(e.g. simulation ID, number of replicates, and number of years to
simulate) and to control output types (plus some more specific
settings). For this overview, we will stick to the defaults:</p>
<pre class="r"><code>sim &lt;- Simulation(Simulation = 2,
                  Years = 50,
                  Replicates = 2,
                  OutIntPop = 50)</code></pre>
<p>For detailed information on this module (or any other), please see
the documentation.</p>
<pre class="r"><code>?Simulation</code></pre>
</div>
<div id="landscape" class="section level2" number="2.2">
<h2><span class="header-section-number">2.2</span> Landscape</h2>
<p><em>RangeShiftR</em> can either import a map from an ASCII raster
file in the ‘Inputs’ folder or generate a random map to use in the
simulation.</p>
<p>For each option, there is a corresponding function to create a
Landscape parameter object</p>
<pre class="r"><code>land &lt;- ImportedLandscape() 
land &lt;- ArtificialLandscape()</code></pre>
<p>Imported landscapes can provide either (binary or continuous) habitat
suitability or land type codes. Furthermore, they can be either patch-
or cell-based. We cover both types of landscapes in the remaining
tutorials.</p>
<p>Artificially generated landscapes can only contain (binary or
continuous) habitat suitability and are always cell-based.</p>
<p>For our example, we define an artificial landscape:</p>
<pre class="r"><code>land &lt;- ArtificialLandscape(Resolution = 10,  # in meters
                            K_or_DensDep = 1500,  # ~ 15 inds/cell
                            propSuit = 0.2,
                            dimX = 129, dimY = 257, 
                            fractal = T, hurst = 0.3,
                            continuous = F)</code></pre>
<!-- ```{r eval=F, echo=F} -->
<!-- library(NLMR) -->
<!-- library(landscapetools) -->
<!-- lsc <- nlm_mpd(ncol = 150, nrow = 150, roughness = 0.9, rand_dev = 2, resolution=10)<=0.45 -->
<!-- lsc@file@nodatavalue <- -9999 -->
<!-- writeRaster(lsc+1, format="ascii", datatype = 'INT1S', -->
<!--                 filename = paste0(dirpath,"Inputs/lsc"), -->
<!--                 overwrite = F) -->
<!-- ``` -->
<!-- ```{r eval=F, echo=F} -->
<!-- land <- ImportedLandscape(LandscapeFile = "lsc.asc",  -->
<!--                           Resolution = 10,  -->
<!--                           Nhabitats = 2,  -->
<!--                           K_or_DensDep = c(0,1500)) -->
<!-- ``` -->
</div>
<div id="demography" class="section level2" number="2.3">
<h2><span class="header-section-number">2.3</span> Demography</h2>
<p>The Demography module contains all the local population dynamics of
your simulated species. Generally there are two types:</p>
<ul>
<li>Unstructured model / non-overlapping generations</li>
<li>Stage-structured model / overlapping generations</li>
</ul>
<p>For the first case, create a simple <code>Demography()</code> module
(the maximum growth rate <code>Rmax</code> is the only required
parameter)</p>
<pre class="r"><code>demo &lt;- Demography(Rmax = 2.2, ReproductionType = 1, PropMales = 0.45)</code></pre>
<p>The option <code>ReproductionType</code> determines the way that
different sexes are considered:</p>
<p>0 = asexual / only female model<br />
1 = simple sexual model<br />
2 = sexual model with explicit mating system</p>
<p>In order to make a stage-structured model, we have to additionally
create a stage-structure sub-module within the Demography module. Here,
we have already defined a Demography object and can use ‘+’ to add the
StageStructure sub-module.</p>
<pre class="r"><code>stg &lt;- StageStructure(Stages = 3,
                      TransMatrix = matrix(c(0,1,0,5.7,.5,.4,3.4,0,.9),nrow = 3),
                      FecDensDep = T,
                      SurvDensDep = T)
demo &lt;- demo + stg</code></pre>
<p>Alternatively, we define the sub-module within the Demography
module:</p>
<pre class="r"><code>demo &lt;- Demography(StageStruct = stg, ReproductionType = 1, PropMales = 0.45)</code></pre>
<p><em>RangeShiftR</em> provides a number of useful functions to explore
the model set-up. For example, we can plot the rates from the transition
matrix:</p>
<pre class="r"><code>plotProbs(stg)</code></pre>
<p><img src="7_RS_intro_files/figure-html/unnamed-chunk-16-1.png" width="672" /></p>
</div>
<div id="dispersal" class="section level2" number="2.4">
<h2><span class="header-section-number">2.4</span> Dispersal</h2>
<p>The dispersal process is modelled wih three sub-processes (see the
schematic figure above): <code>Emigration()</code>,
<code>Transfer()</code> and <code>Settlement()</code>.</p>
<pre class="r"><code>disp &lt;-  Dispersal(Emigration = Emigration(EmigProb = 0.2), 
                   Transfer   = DispersalKernel(Distances = 50),
                   Settlement = Settlement() )</code></pre>
<p>We can use the function <code>plotProbs()</code> to plot various
functional relationships, for example a dispersal kernel with a
stage-dependent mean transfer distance:</p>
<pre class="r"><code>plotProbs(DispersalKernel(Distances = matrix(c(0,1,2,70,50,30),nrow = 3), StageDep = T))</code></pre>
<p><img src="7_RS_intro_files/figure-html/unnamed-chunk-18-1.png" width="672" /></p>
<p>This is using mostly default options. For example, we can change the
settlement condition so that a female individual, that arrives in an
unsuitable cell, will wait for another time step and disperse again,
while the males will die if arriving in an unsuitable cell:</p>
<pre class="r"><code>disp &lt;-  disp + Settlement(SexDep = T, 
                           Settle = matrix(c(0,1,1,0), nrow = 2))</code></pre>
</div>
<div id="initialisation" class="section level2" number="2.5">
<h2><span class="header-section-number">2.5</span> Initialisation</h2>
<p>In order to control the initial distribution of individuals in the
landscape at year <em>0</em>, we set initialisation rules. We choose to
initialise <em>3</em> individuals per habitat cell. Additionally, since
we define a stage-structured model, we have to specify the initial
proportion of stages:</p>
<pre class="r"><code>init &lt;- Initialise(FreeType = 0, 
                   NrCells = 2250,
                   InitDens = 2, 
                   IndsHaCell = 3, 
                   PropStages = c(0,0.7,0.3))
init</code></pre>
<pre><code>##  Initialisation: 
##    InitType = 0 : Free initialisation 
##    of 2250 random suitable cells/patches.
##    InitDens = 2 : 3  individuals per cell/hectare 
##    PropStages = 0 0.7 0.3 
##    InitAge = 2  : Quasi-equilibrium</code></pre>
</div>
<div id="parameter-master" class="section level2" number="2.6">
<h2><span class="header-section-number">2.6</span> Parameter master</h2>
<p>After all settings have been made in their respective modules, we are
ready to combine them to a parameter master object, which is needed to
run the simulation.</p>
<pre class="r"><code>s &lt;- RSsim(simul = sim, land = land, demog = demo, dispersal = disp, init = init)

# Alternative notation:
# s &lt;- RSsim() + land + demo + disp + sim + init + gene</code></pre>
<p>We can check the parameter master (or any single module) for
potential parameter conflicts:</p>
<pre class="r"><code>validateRSparams(s)</code></pre>
<pre><code>## [1] TRUE</code></pre>
</div>
<div id="run-the-simulation" class="section level2" number="2.7">
<h2><span class="header-section-number">2.7</span> Run the
simulation</h2>
<p>Once the parameter master has been defined, we can run the
simulations in the specified RS directory.</p>
<pre class="r"><code>RunRS(s, dirpath)</code></pre>
</div>
</div>
<div id="plot-results" class="section level1" number="3">
<h1><span class="header-section-number">3</span> Plot results</h1>
<p>All results are stored in the Outputs folder. <em>RangeShiftR</em>
provides some in-built functions to access and plot these results. Here,
we plot the abundance and occupancy time series:</p>
<pre class="r"><code>range_df &lt;- readRange(s, dirpath)

# ...with replicates:
par(mfrow=c(1,2))
plotAbundance(range_df)
plotOccupancy(range_df)</code></pre>
<p><img src="7_RS_intro_files/figure-html/unnamed-chunk-24-1.png" width="672" /></p>
<pre class="r"><code># ...with standard deviation:
par(mfrow=c(1,2))
plotAbundance(range_df, sd=T, replicates = F)
plotOccupancy(range_df, sd=T, replicates = F)</code></pre>
<p><img src="7_RS_intro_files/figure-html/unnamed-chunk-24-2.png" width="672" /></p>
<p>Although <em>RangeShiftR</em> provides a number of functions for easy
post-processing and plotting of results, it may also be desirable to
further process the results yourself. As a simple example for such a
workflow, let’s plot the spatial distribution of abundance. For doing
so, we load the data from the population output files and process it
using the <code>terra</code> package:</p>
<pre class="r"><code># read population output file into a dataframe
pop_df &lt;- readPop(s, dirpath)

# Make data frame with number of individuals per output year - for only one repetition (Rep==0):
pop_wide_rep0 &lt;- reshape(subset(pop_df,Rep==0)[,c(&#39;Year&#39;,&#39;x&#39;,&#39;y&#39;,&#39;NInd&#39;)], timevar=&#39;Year&#39;, v.names=c(&#39;NInd&#39;), idvar=c(&#39;x&#39;,&#39;y&#39;), direction=&#39;wide&#39;)

head(pop_wide_rep0)</code></pre>
<pre><code>##     x    y NInd.0 NInd.50
## 1  15  165      3       5
## 2 265 2505      3       3
## 3 425 2435      3       6
## 4 435 1855      3       6
## 5 155 1085      3       4
## 6 515 2335      3       6</code></pre>
<pre class="r"><code># use terra package to make a SpatRaster from the data frame
library(terra)

stack_years_rep0 &lt;- terra::rast(pop_wide_rep0, type=&#39;xyz&#39;)
names(stack_years_rep0) &lt;- c(&#39;Year.0&#39;, &#39;Year.50&#39;)

terra::plot(stack_years_rep0, range = c(0,10), type=&#39;continuous&#39;)</code></pre>
<p><img src="7_RS_intro_files/figure-html/unnamed-chunk-25-1.png" width="672" /></p>
</div>
<div id="references" class="section level1 unnumbered">
<h1 class="unnumbered">References</h1>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-Bocedi2014" class="csl-entry">
Bocedi, G., S. C. F. Palmer, G. Pe’er, R. K. Heikkinen, Y. G. Matsinos,
K. Watts, and J. M. J. Travis. 2014. <span>“RangeShifter: A Platform for
Modelling Spatial Eco-Evolutionary Dynamics and Species’ Responses to
Environmental Changes.”</span> <em>Methods in Ecology and Evolution</em>
5 (4): 388–96. <a
href="https://doi.org/10.1111/2041-210X.12162">https://doi.org/10.1111/2041-210X.12162</a>.
</div>
<div id="ref-Bocedi2021" class="csl-entry">
Bocedi, Greta, Stephen C. F. Palmer, Anne-Kathleen Malchow, Damaris
Zurell, Kevin Watts, and Justin M. J. Travis. 2021. <span>“RangeShifter
2.0: An Extended and Enhanced Platform for Modelling Spatial
Eco-Evolutionary Dynamics and Species’ Responses to Environmental
Changes.”</span> <em>Ecography</em> 44 (10): 1453–62. https://doi.org/<a
href="https://doi.org/10.1111/ecog.05687">https://doi.org/10.1111/ecog.05687</a>.
</div>
<div id="ref-Malchow2021" class="csl-entry">
Malchow, Anne-Kathleen, Greta Bocedi, Stephen C. F. Palmer, Justin M. J.
Travis, and Damaris Zurell. 2021. <span>“RangeShiftR: An r Package for
Individual-Based Simulation of Spatial Eco-Evolutionary Dynamics and
Species’ Responses to Environmental Changes.”</span> <em>Ecography</em>
44 (10): 1443–52. https://doi.org/<a
href="https://doi.org/10.1111/ecog.05689">https://doi.org/10.1111/ecog.05689</a>.
</div>
</div>
</div>

<!DOCTYPE html>
<html>

<br>
<hr />
<div id="footer">
<p>Damaris Zurell 2023 <a href="http://creativecommons.org/licenses/by/4.0/" >(CC BY 4.0)</a>.  </p>
</div>

</html>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
